virtualmachine/vsphere/util.go

```diff
@@ -666,6 +711,13 @@ var cloneFromTemplate = func(vm *VM, dcMo *mo.Datacenter, usableDatastores []str
 	}
 	config.DeviceChange = deviceChangeSpec
 
+	//Resize (increase)/delete existing volumes in VM template
```

Comment:
we are creating the spec from the devices of template and feeding it to the vm deviceChange spec. Please verify if this is ok.

Reply:
Discuss offline..No need to change anything.